ui/src/pages/QuickSetup/AccountSelection.tsx

```diff
@@ -144,8 +144,27 @@ const ProviderOption = (props: {
   const ledger = useLedger();
   const { getName } = usePartyName(party);
 
-  const makeAccountName = (accountInfo: AccountInfo) =>
-    `${getName(party)}-${getName(serviceProvider)}-${accountInfo.accountLabel.replace(/\s+/g, '')}`;
+  const partyName = getName(party);
+  const serviceProviderName = getName(serviceProvider);
+
+  const makeAccountName = useCallback(
+    (accountInfo: AccountInfo): string | undefined => {
+      const name = `${getName(party)}-${getName(
+        serviceProvider
+      )}-${accountInfo.accountLabel.replace(/\s+/g, '')}`;
+
+      if (isHubDeployment) {
+        if (partyName !== party && serviceProviderName !== serviceProvider) {
```

Comment:
if u already know it is a hub deployment, do u still need to check this?

Reply:
yeah - the difference between hub and local is the long party IDs. on local, the party ID is usually human readable anyway 

On Hub, I want to enforce that the `partyName` is different from the actual party ID before using it in the account name label (this could happen, because using `getName(party)` will default to party ID if no other suitable name is found, like if the verifiedIdentities contracts haven't loaded in yet)